src/client/main.go

```diff
@@ -523,54 +528,103 @@ func prepareRemoteNvim(nvrhContext *nvrh_context.NvrhContext, nv *nvim.Nvim) err
 
 	batch := nv.NewBatch()
 
-	// Set $NVRH_SESSION_ID so we can identify the session.
-	batch.Command(fmt.Sprintf(`let $NVRH_SESSION_ID="%s"`, nvrhContext.SessionId))
-	// Let nvim know the channel id so it can send us messages.
-	batch.Command(fmt.Sprintf(`let $NVRH_CHANNEL_ID="%d"`, nv.ChannelID()))
-	// Set $BROWSER so the remote machine can open a browser locally.
-	batch.Command(fmt.Sprintf(`let $BROWSER="%s"`, nvrhContext.BrowserScriptPath))
+	slog.Info("Preparing remote nvim", "mode", mode, "sessionId", nvrhContext.SessionId)
+
+	if mode == "primary" {
+		batch.ExecLua(`
+local session_id = ...
+
+_G._nvrh = {
```

Comment:
Having this on `_G` is a little worrying, since it means other plugins can access it. I think technically all this can be done in one `ExecLua` block, where these globals could then be local, it would just be awkward as more features get added